server/lib/job-queue/handlers/video-channels-sync.ts

```diff
@@ -0,0 +1,265 @@
+import {
+  MUser,
+  MVideoThumbnail
+} from '@server/types/models'
+import { ThumbnailType, VideoChannelImportPayload, VideoChannelSyncState, VideoImportState, VideoPrivacy, VideoState } from '@shared/models'
+import { Hooks } from '@server/lib/plugins/hooks'
+import { JobQueue } from '../job-queue'
+import { updateVideoMiniatureFromUrl } from '@server/lib/thumbnail'
+import { isVideoFileExtnameValid } from '@server/helpers/custom-validators/videos'
+import { hasUnicastURLsOnly, insertIntoDB, processYoutubeSubtitles } from '@server/helpers/youtube-dl/youtube-dl-import-utils'
+import { VideoChannelSyncModel } from '@server/models/video/video-channel-sync'
+import { VIDEO_CHANNEL_MAX_SYNC } from '@server/initializers/constants'
+import { Job } from 'bull'
+import { wait } from '@shared/core-utils'
+
+const processOptions = {
+  maxBuffer: 1024 * 1024 * 30 // 30MB
+}
+
+type ChannelSyncInfo = {
+  total: number
+  alreadyImported: number
+  errors: number
+  successes: number
+}
+
+function formatDateForYoutubeDl (date: Date) {
+  return `${date.getFullYear()}${date.getMonth() + 1}${date.getDate()}`
+}
+
+export async function processVideoChannelsSync () {
+  logger.debug('Running processVideoChannelsSync')
+  if (!CONFIG.IMPORT.SYNCHRONIZATION.ENABLED) {
+    logger.info('Discard channels synchronization as the feature is disabled')
+    return
+  }
+  const syncs: VideoChannelSyncModel[] = await VideoChannelSyncModel.listSyncs()
+  const youtubeDL = await YoutubeDLCLI.safeGet()
+
+  for (const sync of syncs) {
+    try {
+      const syncCreationDate = sync.createdAt
+      sync.state = VideoChannelSyncState.PROCESSING
+      await sync.save()
+      // Format
+      logger.info(`Starting synchronizing "${sync.VideoChannel.name}" with external channel "${sync.externalChannelUrl}"`)
+      const { errors, successes, alreadyImported } = await synchronizeChannel(sync.VideoChannel, sync.externalChannelUrl, {
+        youtubeDL,
+        secondsToWait: 5,
+        lastVideosCount: VIDEO_CHANNEL_MAX_SYNC,
+        after: formatDateForYoutubeDl(syncCreationDate)
+      })
+      if (errors > 0) {
+        sync.state = VideoChannelSyncState.FAILED
+        logger.error(`Finished synchronizing "${sync.VideoChannel.name}" with failures` +
+          ` (failures: ${errors}, imported: ${successes}, ignored because already imported: ${alreadyImported}). Please check the logs.`)
+      } else {
+        sync.state = VideoChannelSyncState.SYNCED
+        logger.info(`Finished synchronizing "${sync.VideoChannel.name}" successfully` +
+          ` (imported: ${successes}, ignored because already imported: ${alreadyImported})`)
+      }
+      await sync.save()
+    } catch (err) {
+      logger.error(`Failed to synchronize channel ${sync.VideoChannel.name}`, { err })
+    }
+  }
+}
+
+export async function processVideoChannelImport (job: Job) {
+  const payload = job.data as VideoChannelImportPayload
+  logger.debug('Running processVideoChannelImport')
```

Comment:
If the import is triggered from a sync object, flag the latter as PROCESSING

Reply:
The more I think of it, the more I believe it should trigger a change to the VideoChannelSync object.

I even think the video-channel-import deserve its own API (`/video-channel-import`) so it can be reused by the CLI and its own UI. And the synchronization creation UI should call this API and we should remove `/video-channel-sync/$1/sync`.

What do you think @Chocobozzz?

Reply:
Agreed! You can do this in this PR

Reply:
Done in 485d19649fb8e224c96579962b54ca4376156f9d

Also I added an option to fully synchronize a channel (i.e. call this API even after having created a synchronization).

Screenshot:
![image](https://user-images.githubusercontent.com/371705/183096760-e4f692cf-d3f4-4725-88f3-0920f0a1d287.jpg)